compiler/src/dotty/tools/dotc/transform/ElimByName.scala

```diff
@@ -159,6 +159,12 @@ class ElimByName extends MiniPhase, InfoTransformer:
       else tree
     }
 
+  override def transformOther(tree: Tree)(using Context): Tree = tree match
+    case tree @ Splice(spliced, tpt) =>
+      assert(dotty.tools.dotc.inlines.Inlines.inInlineMethod)
+      cpy.Splice(tree)(transformAllDeep(spliced), tpt)
```

Comment:
Why does this phase specifically need to handle Splice?
Could the generic mini-phase logic traverse splices (and quotes) automatically instead?

Reply:
> Why does this phase specifically need to handle Splice?


This is to traverse the top-level splice of a macro. Maybe an inline method with a quote might need the same transformation.

> Could the generic mini-phase logic traverse splices (and quotes) automatically instead? 

Yes